### PR TITLE
Add search box props for clear button aria attributes

### LINKED ISSUE #3858: Checkbox: should allow setting ariaLabel/ariaDescription to the clear button
### Bug Report

- __Browser and OS versions__:  Edge + Narrator

#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? (No)

Requested priority: (High) 

Products/sites affected: (Web part)

#### Describe the issue:

https://developer.microsoft.com/en-us/fabric#/components/searchbox
Type something in the searchbox, then press tab to focus on the clear button.
The clear button is focusable but not reading anything meaningful.
We can only set the aria-label for the input but we don't have a way to set either aria-label or aria-description to the clear button. For screen reader users, the button will be completely unusable since only word "button" is read.

#### Actual behavior:

Narrator reads "button" when focus is on the clear button, and we don't have a way to set aria-label for it.

#### Expected behavior:

We can set aria-label to the clear button so it can read meaningful information to screen reader users.

### If applicable, please provide a codepen repro:

https://developer.microsoft.com/en-us/fabric#/components/searchbox

### PR DESCRIPTION
**Pull request by hoovercj**
#### Pull request checklist

- [X] Addresses an existing issue: Fixes #3858
- [X] Include a change request file using `$ npm run change`

#### Description of changes

Add clearButtonAriaLabel and clearButtonAriaDescription props to SearchBox.types.ts and pass them to the icon button in SearchBox.base.tsx

#### Focus areas to test

Ensure that passing the props result in the props being applied to the icon button and read to screen readers when the button has focus.

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.types.ts [hunk 1]
```diff
@@ -64,6 +64,16 @@ export interface ISearchBoxProps extends React.InputHTMLAttributes<HTMLInputElem
...
 
+  /**
+   * The aria label of the clear button for the SearchBox for the benefit of screen readers.
+   */
+  clearButtonAriaLabel?: string;
+
+  /**
+   * The aria description of the clear button for the SearchBox for the benefit of screen readers.
+   */
+  clearButtonAriaDescription?: string;
```
Comment (manishgarg1) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.types.ts]: Hi,

we believe that these props be a part of the clearButtonProps property bag. That would make it easier to evolve the APIs. 

clearButtonProps: {
    ariaLabel;
    ariaDescription:
}

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.types.ts [hunk 2]
```diff
@@ -64,6 +64,16 @@ export interface ISearchBoxProps extends React.InputHTMLAttributes<HTMLInputElem
    */
   ariaLabel?: string;
 
+  /**
+   * The aria label of the clear button for the SearchBox for the benefit of screen readers.
+   */
+  clearButtonAriaLabel?: string;
+
```
Comment (manishgarg1) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.types.ts]: we need to remove this. right ?

Comment (hoovercj) [reply packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.types.ts]: Done.

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.types.ts [hunk 3]
```diff
@@ -64,6 +64,11 @@ export interface ISearchBoxProps extends React.InputHTMLAttributes<HTMLInputElem
    */
   ariaLabel?: string;
 
+  /**
+   * The props for the clear button.
+   */
+  clearButtonProps?: ISearchBoxClearButtonProps;
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.types.ts]: This should just be `IButtonProps` which should support whatever a button supports. That allows the caller to do things like put a `data-automation-id` on it.

Comment (hoovercj) [reply packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.types.ts]: I'd considered that but thought I'd start with the conservative approach of whitelisting properties. Is this the last blocking issue?

Comment (hoovercj) [reply packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.types.ts]: To clarify: I will change this to ibuttonprops tomorrow, I'm just asking if there are other changes needed as well.

Comment (hoovercj) [reply packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.types.ts]: Done.

**Comment by hoovercj** [comment]
Excellent, thanks @manishgarg1. What's the best way for us to know when changes like this and #3961 (if it is merged) are shipped? Is there a regular release schedule or some release tags or something to check?

**Comment by mikewheaton** [comment]
@hoovercj I think releases are automated and come out roughly once a day. You can view the [release log](https://github.com/OfficeDev/office-ui-fabric-react/releases) to see the latest, with notes on what's new.

**Comment by hoovercj** [comment]
Oh excellent, exactly what I was looking for. Thanks @mikewheaton

**Pull request merged**